### PR TITLE
FullyConnectedNode: deserialization fix

### PR DESCRIPTION
**Pull request by v-Golubev**
### Details:
 - *ov::Dimension: added attribute visitor*
 - *FullyConnectedNode: deserialization fix*

### Tickets:
 - *63594*

### CODE REVIEW

**Comment by dmitry-gorokhov** [comment]
@v-Golubev Is it ready for review?

**Comment by v-Golubev** [comment]
@dmitry-gorokhov , Yes, it is ready

**Comment by dmitry-gorokhov** [comment]
@vladislav-volkov Please review the PR

File: ngraph/core/include/openvino/core/dimension.hpp [hunk 1]
```diff
@@ -169,4 +170,24 @@ class OPENVINO_API Dimension {
...
+    AttributeAdapter(ov::Dimension& value) : m_ref(value) {}
+
+    const int64_t& get() override;
+    void set(const int64_t& value) override;
+    operator ov::Dimension&() {
+        return m_ref;
+    }
+
+    OPENVINO_RTTI("AttributeAdapter<ov::Dimension>");
+    BWDCMP_RTTI_DECLARATION;
```
[the remark this answers is not in the document]
Comment (v-Golubev) [reply ngraph/core/include/openvino/core/dimension.hpp]: Deleted, thank you

**Review by vladislav-volkov — APPROVED** [review]

**Review by ilya-lavrenov — APPROVED** [review]

File: ngraph/core/include/openvino/core/dimension.hpp [hunk 2]
```diff
@@ -169,4 +170,23 @@ class OPENVINO_API Dimension {
 /// Inserts the string `?` if `dimension` is dynamic; else inserts `dimension.get_length()`.
 OPENVINO_API
 std::ostream& operator<<(std::ostream& str, const Dimension& dimension);
+
+template <>
+class OPENVINO_API AttributeAdapter<ov::Dimension> : public ValueAccessor<int64_t> {
```
Comment (GlebKazantaev) [inline ngraph/core/include/openvino/core/dimension.hpp:175]: @popovaan please take into account this adapter when you will work on 69092

**Pull request merged**